clang/include/clang/CIR/Dialect/IR/CIROps.td

```diff
@@ -3854,6 +3854,48 @@ def MemCpyOp : CIR_Op<"libc.memcpy"> {
   }];
 }
 
+//===----------------------------------------------------------------------===//
+// MemMoveOp
+//===----------------------------------------------------------------------===//
+
+def MemMoveOp : CIR_Op<"libc.memmove"> {
+  let arguments = (ins Arg<CIR_PointerType, "", [MemWrite]>:$dst,
+                       Arg<CIR_PointerType, "", [MemRead]>:$src,
+                       PrimitiveInt:$len);
+  let summary = "Equivalent to libc's `memmove`";
+  let description = [{
+    Given two CIR pointers, `src` and `dst`, `cir.libc.memmove` will copy `len`
+    bytes from the memory pointed by `src` to the memory pointed by `dst`.
+
+    similiar to `cir.libc.memcpy` but accounts for overlapping memory.
+
+    Examples:
+
+    ```mlir
+      // Copying 2 bytes from one array to a struct:
+      %2 = cir.const #cir.int<2> : !u32i
+      cir.libc.memmove %2 bytes from %arr to %struct : !cir.ptr<!arr> -> !cir.ptr<!struct>
+    ```
+  }];
+
+  let assemblyFormat = [{
+    $len `bytes` `from` $src `to` $dst attr-dict
+    `:` type($len) `` `,` qualified(type($src)) `->` qualified(type($dst))
```

Comment:
Update this line of syntax to:

```suggestion
    `:` qualified(type($dst)) `,` type($len)
```

We don't have to explicitly specify the type of `$dst` since its type is the same as `$src` as the `AllTypeMatch` trait constrains.